user/hasher.go

```diff
@@ -12,17 +15,23 @@ func generateUniqueHash(strings []string, length int) (string, error) {
 
 	if len(strings) > 0 && length > 0 {
 
-		hash := sha1.New()
+		hash := sha256.New()
 
 		for i := range strings {
 			hash.Write([]byte(strings[i]))
 		}
 
-		hash.Write([]byte(strconv.FormatInt(time.Now().Unix(), 10)))
-		//delay just a bit to make sure that we have move on in time
-		time.Sleep(1 * time.Millisecond)
-		hashString := hex.EncodeToString(hash.Sum(nil))
+		max := big.NewInt(9999999999)
+
+		n, err := rand.Int(rand.Reader, max)
 
+		if err != nil {
+			return "", err
+		}
+		hash.Write([]byte(n.String()))
+		//and use unix nano
+		hash.Write([]byte(strconv.FormatInt(time.Now().UnixNano(), 10)))
+		hashString := hex.EncodeToString(hash.Sum(nil))
 		return string([]rune(hashString)[0:length]), nil
 	}
```

Comment:
I think this is fine for now, but we may eventually just want to dump the input strings and use the nano time + a series of random bytes (using the Read function) and sha256 that.  That is, if we don't just go use a third-party library for this.  ;)